Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -477,6 +477,56 @@ void TextureCacheBase::BindTextures()
   }
 }
 
+struct ManualMipmapDetector
+{
+
+  ManualMipmapDetector()
+  {
+    min = { 255, 255, 255, 255 };
+    max = { 0, 0, 0, 0 };
+  }
+
+  void AddLevel(u32 width, u32 height, u32 row_length, const u8* buffer)
+  {
+    ++levels;
+
+    u64 avg[4] = { 0 };
+    for (u32 i = 0; i < height; ++i)
+    {
+      auto* row = buffer;
+      for (u32 j = 0; j < width; ++j)
+      {
+        avg[0] += row[0];
+        avg[1] += row[1];
+        avg[2] += row[2];
+        avg[3] += row[3];
+
+        row += 4;
+      }
+      
+      buffer += row_length;
+    }
+
+    for (u32 c = 0; c < 4; ++c)
+    {
+      u8 cur = avg[c] / (width * height);
```

Comment:
Is there a reason for truncating to u8 here? I'd just pick a double instead. Just a nitpick through...

Reply:
Yeah, this loses some resolution... it should always be between 0 and 255 though. I'll change it to double.